packages/flame_riverpod/example/lib/main.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:async';
+
+import 'package:flame/components.dart' hide Timer;
+import 'package:flame/game.dart';
+import 'package:flame_riverpod/flame_riverpod.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final countingStreamProvider = StreamProvider<int>((ref) {
+  return Stream.periodic(const Duration(seconds: 1), (inc) => inc);
+});
+
+/// Simple provider that returns a [FlameGame] instance.
+final riverpodAwareGameProvider =
+    StateNotifierProvider<RiverpodAwareGameNotifier, FlameGame?>((ref) {
+  return RiverpodAwareGameNotifier();
+});
+
+/// Simple [StateNotifier] that holds the current [FlameGame] instance.
+class RiverpodAwareGameNotifier extends StateNotifier<FlameGame?> {
+  RiverpodAwareGameNotifier() : super(null);
+
+  void set(FlameGame candidate) {
+    state = candidate;
+  }
+}
+
+void main() {
+  runApp(const ProviderScope(child: MyApp()));
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      home: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: const [
+          Expanded(child: FlutterCountingComponent()),
+          Expanded(
+            child: RiverpodGameWidget.initializeWithGame(
+              uninitializedGame: RefExampleGame.new,
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class FlutterCountingComponent extends ConsumerWidget {
+  const FlutterCountingComponent({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final textStyle = Theme.of(context)
+        .textTheme
+        .headlineSmall
+        ?.copyWith(color: Colors.white);
+
+    final stream = ref.watch(countingStreamProvider);
+    return Material(
+      color: Colors.transparent,
+      child: Column(
+        children: [
+          Text('Flutter', style: textStyle),
+          stream.when(
+            data: (value) => Text('$value', style: textStyle),
+            error: (error, stackTrace) => Text('$error', style: textStyle),
+            loading: () => Text('Loading...', style: textStyle),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class RefExampleGame extends FlameGame with HasComponentRef {
```

Comment:
I'd suggest making a custom base class (like ConsumerFlameGame) instead of a mix-in. That's more in line with what Riverpod does.

Reply:
The problem with that is that we would have to do that with all potential game classes, so we would get double the amount of game-classes. For example we would need `ConsumerForge2DGame` too then and that class wouldn't really fit in neither `flame_forge2d` or `flame_riverpod`.

Reply:
Make a flame_forg2d_riverpod package.
That's what hooks_riverpod is

It's not a huge deal though. But it's likely necessary if you wish to support `ref.watch` (cf https://github.com/flame-engine/flame/pull/2367#discussion_r1121339874)

Reply:
> Make a flame_forg2d_riverpod package.

That would end up being way too many extra packages, and we'd have to think about other people creating packages on top of `Flame` too, like Bonfire. So we would need it to be more generic.

> It's not a huge deal though. But it's likely necessary if you wish to support ref.watch (cf https://github.com/flame-engine/flame/pull/2367#discussion_r1121339874)

Couldn't that be done by a mixin? :thinking:

Reply:
It may need to be life-cycle aware, so it might have to be more specific than a mixin reusable everywhere

And in general I'm not sure about the current implementation of said mixin (relying on a static widgetRef field)